onyo/utils.py

```diff
@@ -74,6 +76,30 @@ def get_editor(onyo_root):
     return editor
 
 
+def generate_faux_serial(onyo_root, faux_length=8):
+    """
+    Generate a unique faux serial number and verify that it does not appear in
+    any other asset file name in any directory of the onyo repository.
+    The requested length of faux serial numbers is limited to the interval
+    between 1 and 37.
+
+    Returns on success a unique faux serial number which does not appear in any
+    other asset file name in any directory of the repository.
+    """
+    # check that the requested faux serial number has a valid length
+    if faux_length < 1 or faux_length > 37:
+        raise ValueError("Length of faux serial numbers must be between 1 and 37")
+
+    # generate a new faux serial number while it is in list_of_assets, and
```

Comment:
This comment confused me, specifically the "while it is in list_of_assets". That is bringing the code logic into the comment.

Comments should provide the "straight dope" of what's being achieved.